### PR TITLE
React modal

### PR DESCRIPTION
**Pull request by bobwise**
## What does this PR do?
Implement the React Modal component.

### Associated Issue 
Issue https://github.com/sparkdesignsystem/spark-design-system/issues/1078

### Documentation
 - [X] Update Spark Docs React
 - [X] Update Spark Docs Angular
 - [X] Update Spark Docs Vanilla

### Code
 - [X] Build Component in Spark React
 - [ ] Unit Testing in Spark React with `gulp test-react` (100% coverage, 100% passing)

### Accessibility
- [X] New changes abide by [accessibility requirements](https://sparkdesignsystem.com/docs/accessibility)

### Browser Testing (current version and 1 prior)
  - [X] Google Chrome
  - [ ] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [ ] Mozilla Firefox (Mobile)
  - [ ] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [ ] Microsoft Edge
  - [x] Apple Safari
  - [X] Apple Safari (Mobile)

### Design Review
 - [X] Design reviewed and approved

### CODE REVIEW

**Comment by afebbraro** [comment]
@bobwise  is this something you'd like to go over today at 10am for the React Code Review?

File: src/patterns/components/modals/angular/info/choice.hbs [hunk 1]
```diff
@@ -16,16 +16,15 @@
   <tbody>
     <tr>
       <td class="sprk-u-FontWeight--bold">
-        modaltype
+        modalType
```
Comment (afebbraro) [inline src/patterns/components/modals/angular/info/choice.hbs]: nice fix!

File: src/patterns/components/modals/react/info/choice.hbs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    <tr>
+      <td class="sprk-u-FontWeight--bold">cancelText</td>
+      <td>string</td>
+      <td>This is the text on the secondary action button in the modal. Defaults to "Cancel".</td>
+    </tr>
+
+    <tr>
+      <td class="sprk-u-FontWeight--bold">confirmClick</td>
+      <td>function</td>
+      <td>This function is called when the primary action button is clicked.</td>
```
Comment (afebbraro) [inline src/patterns/components/modals/react/info/choice.hbs]:
```suggestion
      <td>If supplied, this function is called when the primary action button is clicked.</td>
```

File: src/react/projects/spark-react/src/SprkModal/SprkModal.test.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+  expect(wrapper.find('div.sprk-c-ModalMask').length).toBe(1);
+});
+
+it('should accept key events', () => {
+  const wrapper = mount(<SprkModal isVisible={true} />);
+  
+});
+
+// setting focus
```
Comment (afebbraro) [inline src/react/projects/spark-react/src/SprkModal/SprkModal.test.js]: Few empty comments were left in here.

File: src/react/projects/spark-react/src/matchMedia.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+// window.matchMedia is referenced by a utility function in spark-core. This function
```
Comment (afebbraro) [inline src/react/projects/spark-react/src/matchMedia.js]: I think you mentioned that you were removing this file yesterday. Did you mean to leave this in here?

Comment (bobwise) [reply src/react/projects/spark-react/src/matchMedia.js]: yeah I haven't moved it yet. matchMedia is only for the react tests and _setup.js is in the folder with the vanilla tests so they may not be able to be combined.

File: src/react/projects/spark-react/src/SprkModal/SprkModal.test.js (hunk 3 above)
Comment (bobwise) [reply src/react/projects/spark-react/src/SprkModal/SprkModal.test.js]: in progress

**Pull request merged**